server/expression/executor.go

```diff
@@ -262,6 +285,26 @@ func (e Executor) executeFilter(input filters.Value, filter *Filter) (filters.Va
 	return newValue, nil
 }
 
+func (e Executor) convertArrayIntoFilterValue(input types.TypedValue) (filters.Value, error) {
+	var array Array
+	err := json.Unmarshal([]byte(input.Value), &array)
```

Comment:
By using JSON here, we don't lose the types of each value in the array when converting them to `filters.Value` later in the pipeline.